pkg/agent/plugin/nodeattestor/awsiid/iid_test.go

```diff
@@ -38,17 +38,35 @@ const (
 )
 
 var (
+	// This is an insecure, test-only key from RFC 9500, Section 2.1.
+	// It can be used in tests to avoid slow key generation.
+	// See https://pkg.go.dev/crypto/rsa#example-GenerateKey-TestKey
 	signingKeyPEM = []byte(`-----BEGIN RSA PRIVATE KEY-----
-MIIBywIBAAJhAOn4rFLlxONpujl+q/h/kTQzZoqn1nQZbCKEyIPBWO6kkcSqIqON
-aB3i+xyxgZNwkGEkLGRl/Uwasbp7O/sU43wh5ywWp/AG0iFe1RhwMd8LMq5ron6o
-s2eql71hJKsGEwIDAQABAmEAoDa9YcKe8Q68C5TXE8He33z3Ealea3/hET4VxEsI
-p9mfS6kpMQ+qpRSB2aMfVKP1mrAQ4/5TarrG1ZG3T/Mt9Oy1QHbzALvz2XObIvcR
-0cnG353CLQK/nobvWcwAtac5AjEA9k+1a9R6eFaO3grl9yg5XY2+MboV4wjbsDS3
-s4+MivneTPwvK6eHxtoAlYCNOAslAjEA8yy0PJw3TLBK80DryF3r/Q4wd4uYeFhN
-G6EBF0LccLB7GbKpcDHgnNjW/wObx+LXAjBeP4/G6+3U4CIYuojWMvEIaDVPp8m6
-LuiJGxLzxUjc4NF8Gb8e8CLXJxG0IxVmTXUCMQDSPJAG5rgYoUHrVPGEZU8llSLp
-99J2GUFw5Z3f0nprIukKqqA606RxdjdKeoAwLDkCMCptc0jZR3VM4w1wnwvAe0FL
-t61Ol/Q+OqWFX74JwsUU56FqPFm3Y9k7HxDILdedoQ==
+MIIEowIBAAKCAQEAsPnoGUOnrpiSqt4XynxA+HRP7S+BSObI6qJ7fQAVSPtRkqso
```

Comment:
may we start to use testkey.MustRSA for all test like this one? to avoid hardcoded keys?

Reply:
Seems like a good improvement. We will need some more helper methods in `testkey` to represent the keys as PKCS#1 or PKCS#8 PEM blocks to make this more convenient to use in tests. I will add these helpers and update these usages.

Reply:
Actually, taking a closer look at all these tests, it seems that the keys being in PEM format is not really a requirement, just was used as a convenient format to hardcode the keys. So these additional helpers are not really needed.